addons/sale_timesheet/models/account.py

```diff
@@ -49,9 +49,11 @@ def _compute_commercial_partner(self):
         for timesheet in self:
             timesheet.commercial_partner_id = timesheet.task_id.partner_id.commercial_partner_id or timesheet.project_id.partner_id.commercial_partner_id
 
-    @api.depends('so_line.product_id', 'project_id', 'amount')
+    @api.depends('so_line.product_id', 'project_id.billing_type', 'amount')
     def _compute_timesheet_invoice_type(self):
-        for timesheet in self:
+        billed_manually_timesheets = self.filtered(lambda timesheet: timesheet.project_id.allow_billable and timesheet.project_id.billing_type == 'manually')
+        billed_manually_timesheets.timesheet_invoice_type = 'billable_manual'
+        for timesheet in self - billed_manually_timesheets:
```

Comment:
In fact, I am not sure it is really right. 🤔 

Because the option you added is just applied if the task linked has no SOL. 😅
We should probably change some condition in the for each instead of filtering as you did.

Reply:
hey @xavierbol 
what do you mean by `just applied if the task linked has no SOL`
according to the task, it should be applied `regardless if an SOL is set or not`

Reply:
I am discussing with Alexandra about that.

Also, we need to do a migration script for that changes. 🙂

Reply:
> hey @xavierbol what do you mean by `just applied if the task linked has no SOL` according to the task, it should be applied `regardless if an SOL is set or not`

I don't agree with that part: `it should be applied `regardless if an SOL is set or not`

For me, it is only when there is no SOL set on the timesheet we set invoice_type = `billed_manually` or `not billable`